Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -393,3 +393,58 @@ public struct TLSConfiguration {
                                 renegotiationSupport: renegotiationSupport)
     }
 }
+
+// MARK: BestEffortHashable
+extension TLSConfiguration {
+    /// Returns a best effort result of whether two `TLSConfiguration` objects are equal.
+    ///
+    /// The "best effort" stems from the fact that we are checking the pointer to the `keyLogCallback` closure.
+    ///
+    /// - warning: You should probably not use this function. This function can return false-negatives, but not false-positives.
+    public func bestEffortEquals(_ comparing: TLSConfiguration) -> Bool {
+        let isKeyLoggerCallbacksEqual = withUnsafeBytes(of: self.keyLogCallback) { callbackPointer1 in
```

Comment:
given [SR-14520](https://bugs.swift.org/browse/SR-14520) we should probably make this

```swift
let isBitEqual = withUnsafeBytes(of: self) { selfBits in
    withUnsafeBytes(of: comparing) { otherBits in
        return selfBits.elementsEqual(otherBits)
    }
}
```

with a comment liking the bug.

Reply:
@Lukasa wdyt?

Reply:
I still think we should stick with the closure-based version, but yes agree we should link the bug.

Reply:
@Lukasa the problem is that if we do hit the bug, then they will never be equal. The allocations are for reabstraction thunks which aren't equal because they capture (the heap allocs) :P